src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java

```diff
@@ -0,0 +1,275 @@
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import uk.gov.hmcts.ccd.ElasticsearchBaseTest;
+import uk.gov.hmcts.ccd.MockUtils;
+import uk.gov.hmcts.ccd.data.casedetails.SecurityClassification;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.search.CaseSearchResult;
+
+import javax.inject.Inject;
+import java.util.Map;
+
+import static org.elasticsearch.index.query.QueryBuilders.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.doReturn;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.*;
+
+class CaseSearchEndpointESIT extends ElasticsearchBaseTest {
+
+    private static final String POST_SEARCH_CASES = "/searchCases";
+    private static final String CASE_TYPE_ID_PARAM = "ctid";
+
+    @Inject
+    private WebApplicationContext wac;
+    private MockMvc mockMvc;
+
+    @Mock
+    private Authentication authentication;
+
+    @Mock
+    private SecurityContext securityContext;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        doReturn(authentication).when(securityContext).getAuthentication();
+        SecurityContextHolder.setContext(securityContext);
+
+        MockUtils.setSecurityAuthorities(authentication, AUTOTEST1_PUBLIC, AUTOTEST2_PUBLIC);
+
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+    }
+
+    @Nested
+    class CrossCaseTypeSearch {
+
+        @Test // Note that cross case type searches do NOT return case data
+        void shouldReturnAllCasesForAllSpecifiedCaseTypes() throws Exception {
+            String searchRequest = ElasticsearchTestRequest.builder()
+                .query(matchAllQuery())
+                .build().toJsonString();
+
+            MvcResult result = mockMvc.perform(post(POST_SEARCH_CASES)
+                .contentType(MediaType.APPLICATION_JSON)
+                .param(CASE_TYPE_ID_PARAM, caseTypesParam(CASE_TYPE_A, CASE_TYPE_B))
+                .content(searchRequest))
+                .andExpect(status().is(200))
```

Comment:
Is this test working as expected? - when trying this locally I get a 500 response with _Problem getting case type version for 'casetypeA,casetypeB'._